
sched.h is required to compile users of wake_up(),
firmware.c somehow gets sched.h included on newer kernels
but on older kernels this is not happening so include
this on the file itself. This patch should and will
be sent upstream.

--- a/drivers/net/wireless/libertas/firmware.c
+++ b/drivers/net/wireless/libertas/firmware.c
@@ -2,6 +2,7 @@
  * Firmware loading and handling functions.
  */
 
+#include <linux/sched.h>
 #include <linux/firmware.h>
 #include <linux/firmware.h>
 #include <linux/module.h>
